### PR TITLE
planner: support `recommend index set <opt>=<val>`

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: ref #12303

Problem Summary: planner: support `recommend index set <opt>=<val>`

### What changed and how does it work?

planner: support `recommend index set <opt>=<val>`

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/planner/indexadvisor/indexadvisor.go [hunk 1]
```diff
@@ -48,6 +48,13 @@ func AdviseIndexes(ctx context.Context, sctx sessionctx.Context,
...
 	}
 
+	advisorLogger().Info("fill index advisor option")
+	if err := fillOption(sctx, option); err != nil {
+		advisorLogger().Error("fill index advisor option failed", zap.Error(err))
+		return nil, err
+	}
+	advisorLogger().Info("index advisor option filled", zap.Any("option", option))
+
 	advisorLogger().Info("start to recommend indexes")
```
Comment (hawkingrei) [inline pkg/planner/indexadvisor/indexadvisor.go:58]: Can we merge these two logs？

Comment (qw4990) [reply pkg/planner/indexadvisor/indexadvisor.go]: updated

File: pkg/planner/indexadvisor/options.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+	switch opt {
+	case OptMaxNumIndex, OptMaxIndexColumns:
+		x, err := intVal(val)
+		if err != nil {
+			return err
+		}
+		if x <= 0 {
+			return errors.Errorf("invalid value %v for %s", x, opt)
+		}
+		v = fmt.Sprintf("%v", x)
```
Comment (AilinKid) [inline pkg/planner/indexadvisor/options.go:63]: strconv.Itoa?

Comment (qw4990) [reply pkg/planner/indexadvisor/options.go]: updated

**Review by AilinKid — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**